pennylane/templates/subroutines/trotter.py

```diff
@@ -490,3 +494,366 @@ def compute_decomposition(*args, **kwargs):
+        n (int): an integer representing the number of Trotter steps to perform
+        order (int): an integer (:math:`m`) representing the order of the approximation (must be 1 or even)
+        reverse (bool): if true, reverse the order of the operations queued by :code:`qfunc`
+        name (str): an optional name for the instance
+        **non_trainable_kwargs (dict): non-trainable keyword arguments of the first-order expansion function
+    
+    Returns:
+        Callable: a function with the same signature as :code:`qfunc`, when called it queues an instance of 
+            :class:`~.TrotterizedQfunc`
+    
+    **Example**
+
+    .. code-block:: python3
+
+        def first_order_expansion(time, theta, phi, wires, flip=False):
+            "This is the first order expansion (U_1)."
+            qml.RX(time*theta, wires[0])
+            qml.RY(time*phi, wires[1])
+            if flip:
+                qml.CNOT(wires=wires[:2])
+
+        @qml.qnode(qml.device("default.qubit"))
+        def my_circuit(time, theta, phi, num_trotter_steps):
+            qml.trotterize(
+                first_order_expansion,
+                n=num_trotter_steps,
+                order=2,
+            )(time, theta, phi, wires=['a', 'b', 'c'], flip=True)
+            return qml.state()
+        
+    We can visualize the circuit to see the Suzuki-Trotter product formula being applied:
+            
+        >>> time = 0.1
+        >>> theta, phi = (0.12, -3.45)
+        >>> 
+        >>> print(qml.draw(my_circuit, level=3)(time, theta, phi, num_trotter_steps=1))
+        a: ──RX(0.01)──╭●─╭●──RX(0.01)──┤  State
+        b: ──RY(-0.17)─╰X─╰X──RY(-0.17)─┤  State
+        >>>
+        >>>
+        >>> print(qml.draw(my_circuit, level=3)(time, angles, num_trotter_steps=3))
+        a: ──RX(0.00)──╭●─╭●──RX(0.00)───RX(0.00)──╭●─╭●──RX(0.00)───RX(0.00)──╭●─╭●──RX(0.00)──┤  State
+        b: ──RY(-0.06)─╰X─╰X──RY(-0.06)──RY(-0.06)─╰X─╰X──RY(-0.06)──RY(-0.06)─╰X─╰X──RY(-0.06)─┤  State
+
+    """
+
+    @wraps(qfunc)
+    def wrapper(*args, **kwargs):
+        time = args[0]
+        other_args = args[1:]
+        return TrotterizedQfunc(
+            time, *other_args, qfunc=qfunc, n=n, order=order, reverse=reverse, name=name, **kwargs
+        )
+
+    return wrapper
+
+
+@qml.QueuingManager.stop_recording()
+def _recursive_qfunc(time, order, qfunc, wires, reverse, *qfunc_args, **qfunc_kwargs):
+    """Generate a list of operations using the
+    recursive expression which defines the Trotter product.
+    Args:
+        time (float): the evolution 'time'
+        order (int): the order of the Trotter expansion
+        ops (Iterable(~.Operators)): a list of terms in the Hamiltonian
+    Returns:
+        list: the approximation as product of exponentials of the Hamiltonian terms
+    """
+    if order == 1:
+        with qml.tape.QuantumTape() as tape:
+            qfunc(time, *qfunc_args, wires=wires, **qfunc_kwargs)
```

Comment:
```suggestion
        tape = qml.tape.make_qscript(qfunc)(time, *qfunc_args, wires=wires, **qfunc_kwargs)
```
`QuantumTape` was eliminated from our source code, so I'd rather not let it creep back in.